Automattic/sensei
# Integrate store state with stepper navigation

Fixes #3135

### Changes proposed in this Pull Request

* Mark steps as completed based on their actual completion
* When leaving and returning/reloading the page, pick up at the next step
* Allow to navigate back to completed steps
* Don't render until state is loaded (`isFetching` starts with `true`) - this saves a render with default data before the spinner is shown
* Move defaultRoute to QueryStringRouter - makes it a bit clearer, and saves an iteration rendering a different route

### Testing instructions

* Delete the `sensei_setup_wizard_data` to start fresh
* Open Setup Wizard, should show Welcome step
* Complete welcome step
* Welcome step should be marked as completed
* User can navigate between Welcome and Purpose step
* Re-open Setup Wizard (without a `?step=` query parameter) 
* Should open Purpose step, and allow similar navigation between Welcome and Purpose steps

### Not done here

* Block accessing steps not yet available by directly navigating to them, (for example, going to `?step=ready`). — This adds some complexity and issues so if it's not critical we could skip/delay it. Steps not yet available are not clickable, so this is not affecting any real-world use cases.
